core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -227,6 +228,19 @@ public boolean isUsingSecurityToken() {
         return false;
     }
 
+    @RequirePOST
+    public HttpResponse doConfigureRootUrl(StaplerRequest req, StaplerResponse rsp) {
+        Jenkins j = Jenkins.getInstance();
+        j.checkPermission(Jenkins.ADMINISTER);
+
+        String rootUrl = req.getParameter("rootUrl");
```

Comment:
Just replace the current method parameters with

```java
@QueryParameter String rootUrl
```

Reply:
nice feature 👍